src/main/java/org/tugraz/sysds/runtime/data/BasicTensor.java

```diff
@@ -648,8 +667,15 @@ public void compactEmptyBlock() {
 
 	@Override
 	public CacheBlock slice(int rl, int ru, int cl, int cu, CacheBlock block) {
-		// TODO Auto-generated method stub
-		return null;
+		BasicTensor bt = (BasicTensor) block;
+		int[] dims = Arrays.copyOf(_dims, _dims.length);
+		dims[0] = ru - rl + 1;
+		dims[1] = cu - cl + 1;
+		bt.reset(dims, false);
```

Comment:
Why do we copy it if its immediately overwritten - maybe simplify to `dims = new int[]{ru-rl+1, cu-cl+1}`